python/mirage/triton_profiler.py

```diff
@@ -114,39 +110,31 @@ def _run_profile(self, module_path: str, desc: str = "") -> float:
                     raise RuntimeError("Initial kernel test failed")
                 
             # Warmup runs
-            with tqdm(total=self.warmup_iters + self.profile_iters,
-                        desc=f"{desc} (warmup)", 
-                        position=1, 
-                        leave=False) as pbar:
-                
-                for _ in range(self.warmup_iters):
-                    subprocess.run([sys.executable, module_path], 
-                                    stdout=subprocess.DEVNULL if not self.debug else None,
-                                    stderr=subprocess.DEVNULL if not self.debug else None,
-                                    check=False  # 不立即检查返回值，我们自己处理错误
+            for _ in range(self.warmup_iters):
+                subprocess.run([sys.executable, module_path], 
+                                stdout=subprocess.DEVNULL if not self.debug else None,
+                                stderr=subprocess.DEVNULL if not self.debug else None,
+                                check=False  # Ignore errors during warmup
+                            )
+            
+            # Profile runs
+            starter = torch.cuda.Event(enable_timing=True)
+            ender = torch.cuda.Event(enable_timing=True)
+            
+            torch.cuda.synchronize()
+            starter.record()
+            
+            for _ in range(self.profile_iters):
```

Comment:
This implementation actually starting multiple subprocesses to run the kernel, and each subprocess run it once. Can we change to a new design that only launch a single subprocess in which the kernel is executed sequentially for multiple iterations.

Reply:
Done! Generated code will be like:
```
def run_kernel():
    # Warmup iterations
    for _ in range(2):
        device = torch.device('cuda')
        dtensor10133375 = torch.randn((16,4096,), dtype=torch.float16).to(device=device)
        dtensor10133376 = torch.randn((4096,6144,), dtype=torch.float16).to(device=device)
        dtensor10133377 = torch.randn((16,6144,), dtype=torch.float16).to(device=device)
        custom_kernel_1[(128, 1, 1)](dtensor10133377, dtensor10133375, dtensor10133376)
    
    # Profile iterations
    starter = torch.cuda.Event(enable_timing=True)
    ender = torch.cuda.Event(enable_timing=True)
    torch.cuda.synchronize()
    starter.record()
    for _ in range(5):
        device = torch.device('cuda')
        dtensor10133375 = torch.randn((16,4096,), dtype=torch.float16).to(device=device)
        dtensor10133376 = torch.randn((4096,6144,), dtype=torch.float16).to(device=device)
        dtensor10133377 = torch.randn((16,6144,), dtype=torch.float16).to(device=device)
        custom_kernel_1[(128, 1, 1)](dtensor10133377, dtensor10133375, dtensor10133376)
    ender.record()
    torch.cuda.synchronize()
    elapsed_time = starter.elapsed_time(ender) / 5
    print(f'Average time per iteration: {elapsed_time:.3f} ms')

if __name__ == "__main__":
    print("Running kernel...")
    run_kernel()
```

Reply:
Can users can `custom_kernel_1` as a function in the main Python program? Ideally, we want `superoptimize` to return a mugraph that can be directly executed by the caller (see `demo/rms_norm.py` as an example).

Reply:
Yes, since it returns the same class instance just as the cuda backend superoptimizer part

Reply:
Oh not yet

Reply:
1. Support directly calling graph as funcs 
1.1. add execute_mugraph()in generated code 1.2. change __call__ to support several backends 1.3. other relative modify 
2. fix some errors in triton transpiler including mask form and matmul op